### PR TITLE
Update Namespaces/APIs/Documentation for OpenSearch

### PR DESCRIPTION
**Pull request by davidcui-amzn**
### Description
* Replace documentation links with OpenSearch docs
* Update API calls from `_opensearch/_opendistro` to `_plugins`
* Updated namespaces from `com.amazon.opendistroforelasticsearch` to `org.opensearch`
* Updated CI workflow branches for reports-scheduler dependencies
* Add Settings backward compatibility

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zhongnansu** [comment]
Is this PR ready to review? Could you check the CI failure of reports-scheduler?

**Comment by davidcui-amzn** [comment]
> Is this PR ready to review? Could you check the CI failure of reports-scheduler? @davidcui-amzn

Yes, looking now. Failed after I changed the version from beta1 to rc1

**Comment by zhongnansu** [comment]
I noticed we still have some opendistro naming in README, could you also fix that in this PR if it's convenient?
- https://github.com/opensearch-project/dashboards-reports/blob/dev/README.md#opensearch-dashboards-reports-for-open-distro
- https://github.com/opensearch-project/dashboards-reports/blob/dev/README.md#opensearch-dashboards-reports-for-open-distro
Maybe just do a text search for `distro`

**Comment by davidcui-amzn** [comment]
> I noticed we still have some opendistro naming in README, could you also fix that in this PR if it's convenient?
> 
>     * https://github.com/opensearch-project/dashboards-reports/blob/dev/README.md#opensearch-dashboards-reports-for-open-distro
> 
>     * https://github.com/opensearch-project/dashboards-reports/blob/dev/README.md#opensearch-dashboards-reports-for-open-distro
>       Maybe just do a text search for `distro`

Good catch, missed this earlier I only text searched for `openDistro`

File: .github/workflows/dashboards-reports-release-workflow.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ on:
 
 env:
   PLUGIN_NAME: reportsDashboards
-  OPENSEARCH_VERSION: 1.0.0-beta1
-  OPENSEARCH_PLUGIN_VERSION: 1.0.0.0-beta1
+  OPENSEARCH_VERSION: 1.0.0-rc1
```
Comment (joshuali925) [inline .github/workflows/dashboards-reports-release-workflow.yml:10]: i believe we are also using `1.x` branch for OpenSearch Dashboards

File: README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 [![Reports Scheduler CI](https://github.com/opensearch-project/dashboards-reports/workflows/Test%20and%20Build%20Reports%20Scheduler/badge.svg)](https://github.com/opensearch-project/dashboards-reports/actions?query=workflow%3A%22Test+and+Build+Reports+Scheduler%22)
 [![codecov](https://codecov.io/gh/opensearch-project/dashboards-reports/branch/dev/graph/badge.svg?token=FBVYQSZD3B)](https://codecov.io/gh/opensearch-project/dashboards-reports)
-[![Documentation](https://img.shields.io/badge/documentation-blue.svg)](https://opendistro.github.io/for-elasticsearch-docs/docs/kibana/reporting/)
+[![Documentation](https://img.shields.io/badge/documentation-blue.svg)](https://docs-beta.opensearch.org/docs/opensearch-dashboards/reporting/)
 ![PRs welcome!](https://img.shields.io/badge/PRs-welcome!-success)
 
-# OpenSearch Dashboards Reports for Open Distro
+# OpenSearch Dashboards Reports for Open Search
```
Comment (joshuali925) [inline README.md:7]: `OpenSearch Dashboards Reports` should be enough? same comment for other places

File: reports-scheduler/src/main/kotlin/org/opensearch/reportsscheduler/ReportsSchedulerPlugin.kt [hunk 3]
```diff
@@ -80,15 +80,15 @@ import org.opensearch.watcher.ResourceWatcherService
...
  * This class initializes the rest handlers.
  */
 class ReportsSchedulerPlugin : Plugin(), ActionPlugin, JobSchedulerExtension {
 
     companion object {
-        const val PLUGIN_NAME = "opendistro-reports-scheduler"
+        const val PLUGIN_NAME = "opensearch-reports-scheduler"
         const val LOG_PREFIX = "reports"
-        const val BASE_REPORTS_URI = "/_opendistro/_reports"
+        const val BASE_REPORTS_URI = "/_plugins/_reports"
```
Comment (joshuali925) [inline reports-scheduler/src/main/kotlin/org/opensearch/reportsscheduler/ReportsSchedulerPlugin.kt:91]: does this remove the `/_opendistro/_reports` API? I thought we still need to be compatible with them?

File: reports-scheduler/src/main/kotlin/org/opensearch/reportsscheduler/action/DeleteReportDefinitionAction.kt [hunk 4]
```diff
@@ -51,7 +51,7 @@ internal class DeleteReportDefinitionAction @Inject constructor(
     actionFilters,
     ::DeleteReportDefinitionRequest) {
     companion object {
-        private const val NAME = "cluster:admin/opendistro/reports/definition/delete"
+        private const val NAME = "cluster:admin/opensearch/reports/definition/delete"
```
Comment (joshuali925) [inline reports-scheduler/src/main/kotlin/org/opensearch/reportsscheduler/action/DeleteReportDefinitionAction.kt:54]: Seems these should not be changed as security plugin didn't change them yet. ref: https://github.com/opensearch-project/security/issues/1178

**Review by joshuali925 — COMMENTED** [review]
also remove opendistro in these filenames?
https://github.com/opensearch-project/dashboards-reports/blob/odfe-backward-compatibility/dashboards-reports/server/backend/opendistro-notification-plugin.ts

https://github.com/opensearch-project/dashboards-reports/blob/odfe-backward-compatibility/dashboards-reports/server/backend/opendistro-opensearch-reports-plugin.ts

File: reports-scheduler/src/main/kotlin/org/opensearch/reportsscheduler/action/DeleteReportDefinitionAction.kt (hunk 4 above)
Comment (davidcui-amzn) [reply reports-scheduler/src/main/kotlin/org/opensearch/reportsscheduler/action/DeleteReportDefinitionAction.kt:54]: reverting

File: README.md (hunk 2 above)
Comment (davidcui-amzn) [reply README.md:7]: Sure, will change

**Review by joshuali925 — APPROVED** [review]
LGTM

**Comment by zhongnansu** [comment]
Should the `opensearch-job-scheduler-1.0.0.0-beta1.zip` under test resource be renamed to `rc-1`?

**Review by zhongnansu — APPROVED** [review]
lgtm, left one comment

**Comment by davidcui-amzn** [comment]
> Should the `opensearch-job-scheduler-1.0.0.0-beta1.zip` under test resource be renamed to `rc-1`?

Yes it should not be beta1 anymore, good catch. Will rename that and then merge.

**Pull request merged**